### PR TITLE
[evm-price-pusher] Make web3 connections more reslient

### PR DESCRIPTION
**Pull request by ali-bahjati**
If the pusher web3 connection gets dropped then uppon pushing the price it will crash with that error. This PR (1) configures the connection options to be more resilient (reconnect, ...) (2) handles this error explicitly on pushing prices.

This PR will also upgrade web3 packages and adds skipLibCheck to tsconfig.

Update: Added PythContractFactory as suggested. Also changed errors to throw and crash when the error is not handled explicitly. Crashing on unknown behaviors is better than skipping it, because (1) we'll get notified and can investigate (2) a restart might solve the problem.

### CODE REVIEW

File: pyth-evm-price-pusher/src/pusher.ts [hunk 1]
```diff
@@ -39,18 +44,26 @@ export class Pusher {
 
     this.cooldownDuration = config.cooldownDuration;
 
+    this.evmEndpoint = evmEndpoint;
+    this.mnemonic = mnemonic;
+    this.pythContractAddr = pythContractAddr;
+
+    this.pythContract = this.createWeb3PayerPythContract();
```
Comment (jayantk) [inline pyth-evm-price-pusher/src/pusher.ts:51]: suggest making a separate class like PythContractFactory to encapsulate these fields and the creation logic.

File: pyth-evm-price-pusher/src/pusher.ts [hunk 2]
```diff
@@ -149,6 +162,13 @@ export class Pusher {
           throw err;
         }
 
+        if (err.message.includes("connection not open on send")) {
```
Comment (jayantk) [inline pyth-evm-price-pusher/src/pusher.ts:165]: is there a guarantee that these error message strings won't change in the future? this check seems brittle. would be safer to rely on an error type or code or something.

**Review by jayantk — APPROVED** [review]

File: pyth-evm-price-pusher/src/pusher.ts (hunk 2 above)
Comment (ali-bahjati) [reply pyth-evm-price-pusher/src/pusher.ts:165]: I will revisit all the error messages here later. I know it is brittle but I recall that I could not handle it in any other way.

File: pyth-evm-price-pusher/src/pusher.ts (hunk 1 above)
Comment (ali-bahjati) [reply pyth-evm-price-pusher/src/pusher.ts:51]: Good idea! thanks.

**Review by jayantk — APPROVED** [review]

**Pull request merged**